src/Plugin/Field/FieldWidget/LocalistUrlWidget.php

```diff
@@ -0,0 +1,451 @@
+<?php
+
+namespace Drupal\stanford_fields\Plugin\Field\FieldWidget;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\Url as UrlElement;
+use Drupal\Core\Url;
+use Drupal\link\Plugin\Field\FieldWidget\LinkWidget;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'localist_url' widget.
+ *
+ * @FieldWidget(
+ *   id = "localist_url",
+ *   label = @Translation("Localist"),
+ *   field_types = {
+ *     "link"
+ *   }
+ * )
+ */
+class LocalistUrlWidget extends LinkWidget {
+
+  /**
+   * Http Client Service.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $client;
+
+  /**
+   * Caching service.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('http_client'),
+      $container->get('cache.default')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function defaultSettings() {
+    $settings = [
+      'base_url' => '',
+      'select_distinct' => FALSE,
+    ];
+    return $settings + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, ClientInterface $client, CacheBackendInterface $cache) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->client = $client;
+    $this->cache = $cache;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements = parent::settingsForm($form, $form_state);
+    $elements['placeholder_url']['#access'] = FALSE;
+    $elements['placeholder_title']['#access'] = FALSE;
+    $elements['select_distinct'] = FALSE;
+    $elements['select_distinct'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Select Distinct'),
+      '#return_value' => TRUE,
```

Comment:
this shouldn't return true all the time.  You should have a `#default_value` that pull from the settings

Reply:
The default setting is set to FALSE on line 64 above.  According to the documentation, the `#return_value` key indicates what value to return when the checkbox is checked, which I set for the sake of clarity: 

![image](https://user-images.githubusercontent.com/1263892/142474767-4d7666f9-163d-46b5-bbfb-ac55457aa9d8.png)

Am I misunderstanding the documentation?

Reply:
A checkbox will return `1` or `0` which then gets cast to a boolean with https://github.com/SU-SWS/stanford_fields/pull/21#discussion_r752518203 so you don't have to define what it returns
But there is still no `#default_value` here.

Reply:
Gotcha.  This is fixed now.

Reply:
Thanks.